ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```
@@ -0,0 +1,559 @@
+from ansible_collections.arista.avd.plugins.module_utils.eos_designs import AVDFacts
+from functools import cached_property
+from ansible_collections.arista.avd.plugins.module_utils.utils import get
+from ansible_collections.arista.avd.plugins.filter.snmp_hash import hash_passphrase
+from hashlib import sha1
+
+
+class AVDStructConfig(AVDFacts):
+
+    def __init__(self, hostvars=None):
+        self._hash_passphrase = hash_passphrase
+        super().__init__(hostvars)
+
+    @cached_property
+    def _mgmt_interface_vrf(self):
+        return get(self._hostvars, 'mgmt_interface_vrf')
+
+    @cached_property
+    def _mgmt_gateway(self):
+        return get(self._hostvars, 'mgmt_gateway')
+
+    @cached_property
+    def _switch_data(self):
```

1. please call it `_switch` to make it as close to `switch.xxx` in the old templates.
2. I would asume this is required, so we should error out if it is missing. (Just set `required=True`)